### PR TITLE
Update dependencies in coroutines-codelab

### PR DESCRIPTION
**Pull request by osuleymanova**
Kotlin and Gradle Plugin versions updated
apply plugin: 'kotlin-android-extensions' removed
android:exported="true" added in Manifest
typo " expresso" replaced by "espresso"
jcenter() replaced by mavenCentral()

### CODE REVIEW

**Comment by JoseAlcerreca** [comment]
Thanks @osuleymanova !

This particular codelab should use Kotlin 1.5.31 for now, as 1.6 introduces changes that are not covered yet (cc @wojtek-kalicinski @manuelvicnt )

I've hopefully fixed CI, so please merge with `main` to incorporate the changes.

**Comment by JoseAlcerreca** [comment]
@manuelvicnt @wojtek-kalicinski please take a look at the error in `finished_code` module and fix it starting on this PR.

File: coroutines-codelab/build.gradle [hunk 1]
```diff
@@ -20,14 +20,14 @@ buildscript {
     ext {
 
         // global variables
-        kotlin_version = '1.3.61'
+        kotlin_version = '1.5.31'
```
Comment (manuelvicnt) [inline coroutines-codelab/build.gradle:23]: Should we also update the `coroutines_android_version` to 1.5.2? It shouldn't cause any breaking changes

Comment (JoseAlcerreca) [reply coroutines-codelab/build.gradle:23]: Sure. cc @osuleymanova

Comment (osuleymanova) [reply coroutines-codelab/build.gradle:23]: @manuelvicnt 
1.5.2 version caused breaking changes on my side.

**Comment by wojtek-kalicinski** [comment]
CI is failing with: /home/runner/work/kotlin-coroutines/kotlin-coroutines/coroutines-codelab/finished_code/build/tmp/kapt3/stubs/debug/com/example/android/kotlincoroutines/main/TitleDao.java:17: error: Type of the parameter must be a class annotated with @Entity or a collection/array of it.

**Review by manuelvicnt — APPROVED** [review]

**Pull request merged**